platform/linuxbsd/joypad_linux.cpp

```diff
@@ -411,7 +419,17 @@ void JoypadLinux::open_joypad(const char *p_path) {
 			}
 
 			sprintf(uid + String(uid).length(), "%04x%04x%04x%04x%04x%04x", vendor, 0, product, 0, version, 0);
-			input->joy_connection_changed(joy_num, true, name, uid);
+
+			if (inpid.vendor == VALVE_GAMEPAD_VID && inpid.product == VALVE_GAMEPAD_PID) {
+				if (name.begins_with("Microsoft X-Box 360 pad ")) {
+					String idx_str = name.substr(24);
```

Comment:
So this hardcoded number can be replaced by a compile-time constant.

Reply:
How would you suggest I make this compile time constant?

Reply:
Probably `constexpr`, since we can use modern C++ features.

Reply:
I see, I tought he meant using constexpr to calculate the length of the string at compile time to use in substr, i'll cook up the change.

Reply:
You can just add a `#define VALVE_GAMEPAD_NAME_PREFIX "Microsoft X-Box 360 pad "` and do `name.substr(strlen(VALVE_GAMEPAD_NAME_PREFIX))`. I think most compilers are able to optimize out the `strlen()` and, well, if it didn't happen for some reason, it's not the end of the world either.

Reply:
should be ok now!